src/main/webapp/app/exercises/text/assess-new/textblock-feedback-editor/textblock-feedback-editor.component.ts

```diff
@@ -69,4 +69,26 @@ export class TextblockFeedbackEditorComponent implements AfterViewInit {
     didChange(): void {
         this.feedbackChange.emit(this.feedback);
     }
+    /**
+     * Allows the drop of an SGI Element
+     */
+    allowDrop(event: DragEvent) {
+        event.preventDefault();
+    }
+    /**
+     * Connects the SGI with the Feedback of a Submission Element
+     * @param {Event} event - The drop event
+     * the SGI element sent on drag in processed in this method
+     * the corresponding drag method is in StructuredGradingInstructionsAssessmentLayoutComponent
+     */
+    drop(event: any) {
+        event.preventDefault();
+        const data = event.dataTransfer.getData('text');
+        const instruction = JSON.parse(data);
```

Comment:
I think this should be in a service. Do you have a similar code for modeling as well?
I could even imagine a service method like so:
```ts
updateFeedbackWithStructuredGradingInstructionevent(feedback: Feedback, event: any): void
```